engine/dlib/src/test/test_configfile.cpp

```diff
@@ -281,6 +292,100 @@ const TestParam params_long_value[] = {
 
 INSTANTIATE_TEST_CASE_P(LongValue, LongValue, jc_test_values_in(params_long_value));
 
+static int g_ExtensionTestEnabled = 0;
+static int g_ExtensionTestCreated = 0;
+static int g_ExtensionTestDestroyed = 0;
+static const char* g_ExtensionTestString = "world";
+
+static void TestCreate(dmConfigFile::HConfig config)
+{
+    if (!g_ExtensionTestEnabled) return;
+    g_ExtensionTestCreated++;
+}
+static void TestDestroy(dmConfigFile::HConfig config)
+{
+    if (!g_ExtensionTestEnabled) return;
+    g_ExtensionTestDestroyed++;
+}
+static bool TestGetString(dmConfigFile::HConfig config, const char* key, const char* default_value, const char** out)
+{
+    if (!g_ExtensionTestEnabled) return false;
+    if (strstr(key, "ext.") != key)
+        return false;
+
+    if (strcmp("ext.string", key) == 0)
+    {
+        *out = g_ExtensionTestString;
```

Comment:
The reason I added a "Create" / "Destroy" in the extension, is that if the extension needs to return a new string, it needs to be stored somewhere, and thus also garbage collected.

Reply:
Next up is creating an example extension where I will document this further.